### PR TITLE
Prevent wave dashing cripples

### PR DESCRIPTION
**Pull request by hex37**
## Why It's Good For The Game
Makes the weakened condition work as intended in limiting player movement.

## Changelog
:cl:
fix: prevents players with the weakened status from being able to speed up movement by diving repeatedly instead of crawling.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -630,7 +630,7 @@ default behaviour is:
 		var/speed = movement_delay()
 		resting = TRUE
 		var/dir = C.true_dir
-		if(ishuman(src) && (dir))// If true_dir = 0(src isn't moving), doesn't proc
+		if(ishuman(src) && !src.weakened && (dir))// If true_dir = 0(src isn't moving), doesn't proc
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/living.dm:633]: Needless `src.`, same in the code above with `src.client` and `src.stats`.
```suggestion
		if(ishuman(src) && weakened && (_dir))// If true_dir = 0(src isn't moving), doesn't proc
```

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -630,7 +630,7 @@ default behaviour is:
 		var/speed = movement_delay()
 		resting = TRUE
 		var/dir = C.true_dir
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/living.dm:632]: Var with this name already exist.

```suggestion
		var/_dir = C.true_dir
```

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -630,7 +630,7 @@ default behaviour is:
...
-		if(ishuman(src) && (dir))// If true_dir = 0(src isn't moving), doesn't proc
+		if(ishuman(src) && !src.weakened && (dir))// If true_dir = 0(src isn't moving), doesn't proc
 			var/mob/living/carbon/human/H = src
```
Comment (SirRichardFrancis) [inline code/modules/mob/living/living.dm:634]: What if mob isn't human? More runtimes to runtime god.

File: code/modules/mob/living/living.dm (hunk 1 above)
Comment (hex37) [reply code/modules/mob/living/living.dm:633]: If you're in a situation where you actually need to use src to differentiate vars you're slready writing bad code. This just improves readability and has no effect on speed at compile time.

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (hex37) [reply code/modules/mob/living/living.dm:632]: Unrelated to my change.

File: code/modules/mob/living/living.dm (hunk 3 above)
Comment (hex37) [reply code/modules/mob/living/living.dm:634]: ishuman ensures the mob is human

**Comment by Kegdo** [comment]
so, is this just a one-line change? i cant really see because of the reviews

**Comment by hex37** [comment]
> so, is this just a one-line change? i cant really see because of the reviews

yes.

**Review by Firefox13 — APPROVED** [review]

**Pull request merged**